### PR TITLE
Add support for cephfs helm chat

### PR DESCRIPTION
**Pull request by Madhu-1**
This PR adds the support for cephfs helm chat

logs
```
[🎩]mrajanna@localhost ceph-csi $]hlm install  --name "ceph-csi-cephfs" deploy/cephfs/helm/
NAME:   ceph-csi-cephfs
LAST DEPLOYED: Thu Feb 21 10:42:40 2019
NAMESPACE: default
STATUS: DEPLOYED

RESOURCES:
==> v1/ServiceAccount
NAME                         SECRETS  AGE
ceph-csi-cephfs-attacher     1        1s
ceph-csi-cephfs-nodeplugin   1        1s
ceph-csi-cephfs-provisioner  1        1s

==> v1/ClusterRole
NAME                         AGE
ceph-csi-cephfs-attacher     1s
ceph-csi-cephfs-nodeplugin   1s
ceph-csi-cephfs-provisioner  1s

==> v1/ClusterRoleBinding
NAME                         AGE
ceph-csi-cephfs-attacher     1s
ceph-csi-cephfs-nodeplugin   1s
ceph-csi-cephfs-provisioner  1s

==> v1/Service
NAME                         TYPE       CLUSTER-IP    EXTERNAL-IP  PORT(S)    AGE
ceph-csi-cephfs-attacher     ClusterIP  10.233.6.190  <none>       12345/TCP  1s
ceph-csi-cephfs-provisioner  ClusterIP  10.233.45.41  <none>       12345/TCP  1s

==> v1beta2/DaemonSet
NAME                        DESIRED  CURRENT  READY  UP-TO-DATE  AVAILABLE  NODE SELECTOR  AGE
ceph-csi-cephfs-nodeplugin  3        3        0      3           0          <none>         1s

==> v1beta1/StatefulSet
NAME                         DESIRED  CURRENT  AGE
ceph-csi-cephfs-attacher     1        1        1s
ceph-csi-cephfs-provisioner  1        1        1s

==> v1/Pod(related)
NAME                              READY  STATUS             RESTARTS  AGE
ceph-csi-cephfs-nodeplugin-7nhgw  0/2    ContainerCreating  0         0s
ceph-csi-cephfs-nodeplugin-bhtpw  0/2    ContainerCreating  0         1s
ceph-csi-cephfs-nodeplugin-x9jsw  0/2    ContainerCreating  0         0s
ceph-csi-cephfs-attacher-0        0/1    ContainerCreating  0         1s
ceph-csi-cephfs-provisioner-0     0/2    ContainerCreating  0         0s


NOTES:
Examples on how to configure a storage class and start using the driver are here:
https://github.com/ceph/ceph-csi/tree/csi-v1.0/examples/cephfs

[🎩︎]mrajanna@localhost ceph-csi $]kube get  po
NAME                               READY   STATUS    RESTARTS   AGE
ceph-csi-cephfs-attacher-0         1/1     Running   0          13s
ceph-csi-cephfs-nodeplugin-7nhgw   2/2     Running   0          12s
ceph-csi-cephfs-nodeplugin-bhtpw   2/2     Running   0          13s
ceph-csi-cephfs-nodeplugin-x9jsw   2/2     Running   0          12s
ceph-csi-cephfs-provisioner-0      2/2     Running   0          12s
[🎩︎]mrajanna@localhost ceph-csi $]hlm list
NAME           	REVISION	UPDATED                 	STATUS  	CHART                	APP VERSION	NAMESPACE
ceph-csi-cephfs	1       	Thu Feb 21 10:42:40 2019	DEPLOYED	ceph-csi-cephfs-0.4.0	1.0.0      	default  
[🎩︎]mrajanna@localhost ceph-csi $]hlm delete ceph-csi-cephfs
release "ceph-csi-cephfs" deleted

```

### CODE REVIEW

**Comment by Madhu-1** [comment]
@rootfs @kfox1111 PTAL

**Comment by Madhu-1** [comment]
@gman0 PTAL

File: deploy/rbd/helm/README.md [hunk 1]
```diff
@@ -4,10 +4,10 @@ The ceph-csi-rbd chart adds rbd volume support to your cluster.
 
 ## Install Chart
 
-To install the Chart into your Kubernetes cluster :
+To install the Chart into your Kubernetes cluster
 
 ```bash
-helm install --namespace "ceph-csi-rbd" --name "ceph-csi-rbd" ceph-csi/ceph-csi-rbd
+helm install --name "ceph-csi-rbd" ceph-csi/ceph-csi-rbd
```
Comment (kfox1111) [inline deploy/rbd/helm/README.md]: Why remove the namespace? This puts it in "default"

File: deploy/cephfs/helm/templates/nodeplugin-clusterrole.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    resources: ["namespaces"]
+    verbs: ["get", "list"]
+  - apiGroups: [""]
+    resources: ["persistentvolumes"]
+    verbs: ["get", "list", "watch", "update"]
+  - apiGroups: ["storage.k8s.io"]
+    resources: ["volumeattachments"]
+    verbs: ["get", "list", "watch", "update"]
+  - apiGroups: [""]
+    resources: ["configmaps"]
```
Comment (kfox1111) [inline deploy/cephfs/helm/templates/nodeplugin-clusterrole.yaml]: Can you please double check the permssions for cephfs? rbd needs configmap reading, but I'm not sure cephfs does.

File: deploy/rbd/helm/README.md (hunk 1 above)
Comment (Madhu-1) [reply deploy/rbd/helm/README.md]: current templates in kubernetes folder deploy everything in the default namespace, to keep consistency between both I deleted  helm deployment in a different namespace

File: deploy/cephfs/helm/templates/nodeplugin-clusterrole.yaml (hunk 2 above)
Comment (Madhu-1) [reply deploy/cephfs/helm/templates/nodeplugin-clusterrole.yaml]: taken this from https://github.com/ceph/ceph-csi/blob/csi-v1.0/deploy/cephfs/kubernetes/csi-nodeplugin-rbac.yaml#L25-L27

**Comment by Madhu-1** [comment]
> All in all, this looks really good. Thanks for contributing it.
> 
> There are two files removed that I think you didn't intend to:
> deploy/rbd/helm/templates/provisioner-role.yaml
> deploy/rbd/helm/templates/provisioner-rolebinding.yaml

as we are creating cluster-role  and binding  cluster-role to  a service-account, a normal  role  is not required, tested this and everything  is working fine

File: deploy/rbd/helm/README.md (hunk 1 above)
Comment (kfox1111) [reply deploy/rbd/helm/README.md]: I don't think installing it in "default" is a good recommendation for the users. Please undo.

File: deploy/cephfs/helm/templates/nodeplugin-clusterrole.yaml (hunk 2 above)
Comment (kfox1111) [reply deploy/cephfs/helm/templates/nodeplugin-clusterrole.yaml]: Ok. I'll talk with rootfs about it. But I don't think that should block this pr.

**Comment by kfox1111** [comment]
> > deploy/rbd/helm/templates/provisioner-role.yaml
> > deploy/rbd/helm/templates/provisioner-rolebinding.yaml
> 
> as we are creating cluster-role and binding cluster-role to a service-account, a normal role is not required, tested this and everything is working fine

You tested out the rbd driver without those two files and provisioning rbd volumes still worked?

**Comment by Madhu-1** [comment]
> > > deploy/rbd/helm/templates/provisioner-role.yaml
> > > deploy/rbd/helm/templates/provisioner-rolebinding.yaml
> > 
> > 
> > as we are creating cluster-role and binding cluster-role to a service-account, a normal role is not required, tested this and everything is working fine
> 
> You tested out the rbd driver without those two files and provisioning rbd volumes still worked?

here are the logs
```
[🎩]mrajanna@localhost ceph-csi $]kube creahlm install --namespace "ceph-csi-rbd" --name "ceph-csi-rbd" deploy/rbd/helm/
NAME:   ceph-csi-rbd
LAST DEPLOYED: Fri Feb 22 10:08:42 2019
NAMESPACE: ceph-csi-rbd
STATUS: DEPLOYED

RESOURCES:
==> v1beta1/StatefulSet
NAME                      DESIRED  CURRENT  AGE
ceph-csi-rbd-attacher     1        1        1s
ceph-csi-rbd-provisioner  1        1        1s

==> v1/Pod(related)
NAME                           READY  STATUS             RESTARTS  AGE
ceph-csi-rbd-nodeplugin-9jsfb  0/2    ContainerCreating  0         0s
ceph-csi-rbd-nodeplugin-rfsmp  0/2    ContainerCreating  0         0s
ceph-csi-rbd-nodeplugin-xbp9j  0/2    ContainerCreating  0         0s
ceph-csi-rbd-attacher-0        0/1    ContainerCreating  0         0s
ceph-csi-rbd-provisioner-0     0/3    ContainerCreating  0         0s

==> v1/ServiceAccount
NAME                      SECRETS  AGE
ceph-csi-rbd-attacher     1        1s
ceph-csi-rbd-nodeplugin   1        1s
ceph-csi-rbd-provisioner  1        1s

==> v1/ClusterRole
NAME                      AGE
ceph-csi-rbd-attacher     1s
ceph-csi-rbd-nodeplugin   1s
ceph-csi-rbd-provisioner  1s

==> v1/ClusterRoleBinding
NAME                      AGE
ceph-csi-rbd-attacher     1s
ceph-csi-rbd-nodeplugin   1s
ceph-csi-rbd-provisioner  1s

==> v1/Service
NAME                      TYPE       CLUSTER-IP    EXTERNAL-IP  PORT(S)    AGE
ceph-csi-rbd-attacher     ClusterIP  10.233.6.108  <none>       12345/TCP  1s
ceph-csi-rbd-provisioner  ClusterIP  10.233.43.73  <none>       12345/TCP  1s

==> v1beta2/DaemonSet
NAME                     DESIRED  CURRENT  READY  UP-TO-DATE  AVAILABLE  NODE SELECTOR  AGE
ceph-csi-rbd-nodeplugin  3        3        0      3           0          <none>         1s


NOTES:
Examples on how to configure a storage class and start using the driver are here:
https://github.com/ceph/ceph-csi/tree/csi-v1.0/examples/rbd


[🎩︎]mrajanna@localhost rbd $]kube get po -nceph-csi-rbd
NAME                            READY   STATUS    RESTARTS   AGE
ceph-csi-rbd-attacher-0         1/1     Running   0          5m5s
ceph-csi-rbd-nodeplugin-9jsfb   2/2     Running   1          5m5s
ceph-csi-rbd-nodeplugin-rfsmp   2/2     Running   1          5m5s
ceph-csi-rbd-nodeplugin-xbp9j   2/2     Running   1          5m5s
ceph-csi-rbd-provisioner-0      3/3     Running   0          5m5s
[🎩︎]mrajanna@localhost rbd $]kube get pvc
NAME      STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
rbd-pvc   Bound    pvc-588b1039-365c-11e9-9088-5254003a618c   1Gi        RWO            csi-rbd        47s
```

File: deploy/rbd/helm/README.md (hunk 1 above)
Comment (Madhu-1) [reply deploy/rbd/helm/README.md]: undo done

**Comment by kfox1111** [comment]
I think it will start, but it won't function properly without permissions to read/write the configmaps.

The rbd helm rbac is a little different then the static manifests. The security is tighter as it only has permissions to the configmaps in the specific namespace rather then for the entire cluster. Its a change I wouild like to propose to the static manifests too, but haven't had time yet.

**Comment by Madhu-1** [comment]
> I think it will start, but it won't function properly without permissions to read/write the configmaps.
> 

means volume mount won't work as expected?

> The rbd helm rbac is a little different then the static manifests. The security is tighter as it only has permissions to the configmaps in the specific namespace rather then for the entire cluster. Its a change I wouild like to propose to the static manifests too, but haven't had time yet.

i am not aware of this issue.

to make this PR completed, do i  need to bring back `
deploy/rbd/helm/templates/provisioner-role.yaml deploy/rbd/helm/templates/provisioner-rolebinding.yaml
` files?

**Comment by kfox1111** [comment]
Yeah, I don't think it will work properly to provision the volumes without the role.

I think it best if the PR only added the new chart, not touched the other chart. Its mergeable in its current state without the rbd changes.

**Comment by Madhu-1** [comment]
@kfox1111 reverted back  the  changes for role and role binding

**Comment by rootfs** [comment]
thanks @kfox1111

**Pull request merged**